modules/tftp/server.rb

```diff
@@ -90,21 +107,86 @@ def pxe_default
     end
 
     def pxeconfig_file(mac)
-      ["#{pxeconfig_dir}/menu.lst.01" + mac.delete(':').upcase, "#{pxeconfig_dir}/01-" + mac.tr(':', '-').upcase]
+      ["#{pxeconfig_dir}/menu.lst.01" + mac.delete(':').upcase, "#{pxeconfig_dir}/01-" + dashed_mac(mac).upcase]
     end
   end
 
   class Pxegrub2 < Server
-    def pxeconfig_dir
-      "#{path}/grub2"
+    def bootloader_path(os, version, arch)
+      unless (bootloader_universe = Proxy::TFTP::Plugin.settings.bootloader_universe)
+        logger.debug "TFTP: bootloader universe not configured."
+        return
+      end
+
+      bootloader_path = File.join(bootloader_universe, 'pxegrub2', os, version, arch)
+
+      logger.debug "TFTP: Checking bootloader universe for suitable bootloader directory for"
+      logger.debug "TFTP:   * Operating system: #{os}"
+      logger.debug "TFTP:   * Version: #{version}"
+      logger.debug "TFTP:   * Architecture: #{arch}"
```

Comment:
This is personal preference, but I like log statements to be concise and ideally single line. If there are multiple requests, it may be very hard to track them all.

Something like:
```suggestion
      logger.debug "TFTP: Checking bootloader universe for suitable bootloader directory for OS '#{os}' version '#{version}' (#{arch})"
```